### ISSUE TITLE
Character level styles have a bug

### ISSUE THREAD

**Issue by sasensi** [Agamnentzar/ag-psd#63]
I started playing with character level styles (named `styleRuns` in the library input syntax) and discovered some buggy edge cases.
I think that there is a bug that has to do with the text content length, relatively to the `styleRuns` cumulated length.
What I observe is that, in order to get the expected result, you have to add 1 to the last `styleRuns` item's length.

Taking as example, a text with content `ab` that has a length of `2`, if you try to style each letter separately by adding an item for each one in the `styleRuns` array, with a length of 1, you will get surprising result: the letter `b` will have the same style than `a`.

Based on you readme example, here are some inputs to reproduce the bug:
- last letter style is not applied properly:
```
{
  width: 200,
  height: 200,
  children: [{
    name: 'text layer',
    text: {
      text: 'ab',
      transform: [1, 0, 0, 1, 50, 50],
      style: { font: { name: 'ArialMT' }, fontSize: 30 },
      styleRuns: [
        {
          length: 1,
          style: { fillColor: { r: 255, g: 0, b: 0 } },
        },
        {
          length: 1,
          style: { fillColor: { r: 0, g: 0, b: 255 } },
        },
      ],
    },
  }],
}
```

- photoshop crashes when opening the file:
```
{
  width: 200,
  height: 200,
  children: [{
    name: 'text layer',
    text: {
      text: 'a',
      transform: [1, 0, 0, 1, 50, 50],
      style: { font: { name: 'ArialMT' }, fontSize: 30 },
      styleRuns: [
        {
          length: 1,
          style: { fillColor: { r: 255, g: 0, b: 0 } },
        },
      ],
    },
  }],
}
```

- expected result is produced when adding 1 to the length of the last styleRuns item (this is actually also what we get when reading the file using the library):
```
{
  width: 200,
  height: 200,
  children: [{
    name: 'text layer',
    text: {
      text: 'ab',
      transform: [1, 0, 0, 1, 50, 50],
      style: { font: { name: 'ArialMT' }, fontSize: 30 },
      styleRuns: [
        {
          length: 1,
          style: { fillColor: { r: 255, g: 0, b: 0 } },
        },
        {
          length: 2,
          style: { fillColor: { r: 0, g: 0, b: 255 } },
        },
      ],
    },
  }],
}
```

**Comment by Agamnentzar** [comment]
Should be fixed in version `12.2.0`

Photoshop adds trailing \r to the end of text string and I forgot to adjust run lengths to match it. Now it should also fix the lengths for you when writing PSD if you accidentally put lengths that don't match text length.

**Comment by sasensi** [comment]
Awesome, I confirm that this now works as expected, thank you a lot !
